File: cpp/0078子集.cpp
#include<vector>

using namespace std;
vector<vector<int>> res;
vector<int> temp;


void BackTracking(int start,vector<int> &nums)
{
    for (int i = start; i < nums.size(); ++i)
    {
        temp.push_back(nums[i]);
        res.push_back(temp);
        BackTracking(i + 1, nums);
        temp.pop_back();
    }
}

vector<vector<int>> subsets(vector<int> &nums)
{
    res.push_back(temp);
    BackTracking(0, nums);
    return res;
}

File: python/main/0078子集.py


def subsets(nums:list)->list:
    res=[]
    temp=[]
    def _backTracking(start:int,nums:list):

        for i in range(start,len(nums)):
            temp.append(nums[i])
            res.append(temp.copy())
            _backTracking(i + 1, nums)
            temp.pop()
        

    res.append(temp)
    _backTracking(0, nums)
    
    return res
